quickwit/quickwit-metastore/src/tests/mod.rs

```diff
@@ -45,6 +48,18 @@ pub trait DefaultForTest {
     async fn default_for_test() -> Self;
 }
 
+#[non_exhaustive]
+pub enum MetastoreShardsTestInput {
+    FileBackedIndex,
+    #[cfg(feature = "postgres")]
+    PostgresIndex(Pool<Postgres>),
+}
+
+#[async_trait]
+pub trait DefaultForShard {
```

Comment:
This is my solution to solve the problem of having a single test function for both the postgres and the filebacked metastore. Ideally, we should modify the function `default_for_test` to accept an additional input paramater. But it is being used in a lot of test functions so I did not want to proceed with that change without your approval @guilload .